lexicon/providers/cloudflare.py

```diff
@@ -74,21 +74,31 @@ def _list_records(self, rtype=None, name=None, content=None):
         if content:
             filter_obj['content'] = content
 
-        payload = self._get(
-            '/zones/{0}/dns_records'.format(self.domain_id), filter_obj)
-
         records = []
-        for record in payload['result']:
-            processed_record = {
-                'type': record['type'],
-                'name': record['name'],
-                'ttl': record['ttl'],
-                'content': record['content'],
-                'id': record['id']
-            }
-            records.append(processed_record)
+        while True:
+            payload = self._get(
+                '/zones/{0}/dns_records'.format(self.domain_id), filter_obj)
+
+            LOGGER.debug("payload: %s", payload)
+
+            for record in payload['result']:
+                processed_record = {
+                    'type': record['type'],
+                    'name': record['name'],
+                    'ttl': record['ttl'],
+                    'content': record['content'],
+                    'id': record['id']
+                }
+                records.append(processed_record)
+
+            pages = payload['result_info']['total_pages']
```

Comment:
So during the first iteration of the loop, the "page" parameter is not set. Do you confirm that "page" and "pages" are still present in the response in this case, either with a paginated and non paginated result?

Reply:
As far as I could see, the result is always paginated.
This means `total_pages` and `page` are present even if `page=1` is not specified.

I tested the code by setting `per_page: 10` and retrieving all ~80 records from my CF zone.

Reply:
> On top of my inline comment, is it possible for you to register a new set of records for the tests, in order to take into account the pagination logic for fresh data from the live API?

It doesn't look like much fun to be honest :-D but I can have a look into that.

Any pointer to get me started would be great.